api/metric_metadata_api.go

```diff
@@ -16,86 +16,66 @@
 // Refer to the doc
 // https://docs.google.com/a/squareup.com/document/d/1k0Wgi2wnJPQoyDyReb9dyIqRrD8-v0u8hz37S282ii4/edit
 // for the terminology.
+
 package api
 
 import "github.com/square/metrics/inspect"
 
-// API is the set of public methods exposed by the indexer library.
-type API interface {
+type MetricMetadataConfig struct {
+	// Location of conversion rules. All *.yaml files in here will be loaded.
+	//TODO(cchandler): Move this into the util package along with
+	//other rules + graphite stuff.
+	ConversionRulesPath string `yaml:"conversion_rules_path"`
+
+	// Database configurations
+	// mostly cassandra configurations from
+	// https://github.com/gocql/gocql/blob/master/cluster.go
+	Hosts    []string `yaml:"hosts"`
+	Keyspace string   `yaml:"keyspace"`
+}
+
+type MetricMetadataAPI interface {
 	// AddMetric adds the metric to the system.
 	AddMetric(metric TaggedMetric) error
 	// Bulk metrics addition
 	AddMetrics(metric []TaggedMetric) error
-
 	// RemoveMetric removes the metric from the system.
 	RemoveMetric(metric TaggedMetric) error
-
-	// Convert the given tag-based metric name to graphite metric name,
-	// using the configured rules. May error out.
-	ToGraphiteName(metric TaggedMetric) (GraphiteMetric, error)
-
-	// Converts the given graphite metric to the tag-based meric,
-	// using the configured rules. May error out.
-	ToTaggedName(metric GraphiteMetric) (TaggedMetric, error)
-
 	// For a given MetricKey, retrieve all the tagsets associated with it.
 	GetAllTags(metricKey MetricKey) ([]TagSet, error)
-
 	// GetAllMetrics returns all metrics managed by the system.
 	GetAllMetrics() ([]MetricKey, error)
-
 	// For a given tag key-value pair, obtain the list of all the MetricKeys
 	// associated with them.
 	GetMetricsForTag(tagKey, tagValue string) ([]MetricKey, error)
 }
 
-// Configuration is the struct that tells how to instantiate a new copy of an API.
-type Config struct {
-	// Location of conversion rules. All *.yaml files in here will be loaded.
-	ConversionRulesPath string `yaml:"conversion_rules_path"`
-
-	// Database configurations
-	// mostly cassandra configurations from
-	// https://github.com/gocql/gocql/blob/master/cluster.go
-	Hosts    []string `yaml:"hosts"`
-	Keyspace string   `yaml:"keyspace"`
-}
-
-// ProfilingAPI wraps an ordinary API and also records profiling metrics to a given Profiler object.
-type ProfilingAPI struct {
-	Profiler *inspect.Profiler
-	API      API
+type ProfilingMetricMetadataAPI struct {
+	Profiler       *inspect.Profiler
```

Comment:
Interesting how we have the profiler inline on the struct, whereas above in `api/backend/multibackend.go` we pull it from `request.Profiler`